### PR TITLE
feat(category): define allowed mime types

### PR DESCRIPTION
**Pull request by Yelinz**
optionally restrict which mime types are allowed per category

### CODE REVIEW

File: alexandria/core/migrations/0015_category_allowed_mime_types.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        migrations.AlterField(
+            model_name='tagsynonymgroup',
+            name='modified_by_group',
+            field=models.CharField(blank=True, max_length=255, null=True, verbose_name='modified by group'),
+        ),
+        migrations.AlterField(
+            model_name='tagsynonymgroup',
+            name='modified_by_user',
+            field=models.CharField(blank=True, max_length=150, null=True, verbose_name='modified by user'),
+        ),
```
Comment (anehx) [inline alexandria/core/migrations/0015_category_allowed_mime_types.py:78]: What happened here?

File: alexandria/core/serializers.py [hunk 2]
```diff
@@ -226,6 +226,17 @@ def validate(self, *args, **kwargs):
...
+
+        category = models.Document.objects.get(
+            pk=self.initial_data.get("document")["id"]
+        ).category
+        if (
+            category.allowed_mime_types is not None
+            and len(category.allowed_mime_types)
+            and mime_type not in category.allowed_mime_types
+        ):
+            raise ValidationError(f"File type {mime_type} is not allowed in category {category}.")
```
Comment (anehx) [inline alexandria/core/serializers.py:238]: Translate

File: alexandria/core/migrations/0015_category_allowed_mime_types.py (hunk 1 above)
Comment (Yelinz) [reply alexandria/core/migrations/0015_category_allowed_mime_types.py:78]: `verbose_name` of the `BaseModel` for modified by had a copy paste error. Migrations werent ran to reflect the change of that text.

Comment (winged) [reply alexandria/core/migrations/0015_category_allowed_mime_types.py]: I'd like to see that fixed in a separate commit, even if it implies a separate migration

File: alexandria/core/serializers.py (hunk 2 above)
Comment (Yelinz) [reply alexandria/core/serializers.py]: Done

File: alexandria/core/serializers.py [hunk 3]
```diff
@@ -206,26 +207,34 @@ def validate(self, *args, **kwargs):
...
+        elif variant == models.File.Variant.ORIGINAL and validated_data.get("original"):
             raise ValidationError(
-                f'"original" must not be set for variant "{variant}".'
+                _(f'"original" must not be set for variant "{variant}".')
             )
 
         mime_type = validated_data["content"].content_type
         if mime_type == "application/octet-stream":
-            guess, _ = guess_type(validated_data["name"])
+            guess, encoding = guess_type(validated_data["name"])
```
Comment (czosel) [inline alexandria/core/serializers.py:222]: i've seen people use `__` if `_` is already used, but I'm not sure if I like it or not :-D

File: alexandria/core/migrations/0015_category_allowed_mime_types.py (hunk 1 above)
Comment (Yelinz) [reply alexandria/core/migrations/0015_category_allowed_mime_types.py]: spllit into a seperate commit & migration

File: alexandria/core/serializers.py [hunk 4]
```diff
@@ -206,26 +208,34 @@ def validate(self, *args, **kwargs):
...
-            )
-
-        if (
-            variant := validated_data.get("variant")
-        ) == models.File.Variant.ORIGINAL and validated_data.get("original"):
+        variant = validated_data.get("variant")
+        if variant != models.File.Variant.ORIGINAL and not validated_data.get(
+            "original"
+        ):
+            raise ValidationError(_(f'"original" must be set for variant "{variant}".'))
```
Comment (anehx) [inline alexandria/core/serializers.py:215]: An `f` string inside gettext is not gonna work. You can do that this way:
```suggestion
            raise ValidationError(_('"original" must be set for variant "%(variant)s".') % {"variant": variant})
```

File: alexandria/core/serializers.py [hunk 5]
```diff
@@ -206,26 +208,34 @@ def validate(self, *args, **kwargs):
...
+        elif variant == models.File.Variant.ORIGINAL and validated_data.get("original"):
             raise ValidationError(
-                f'"original" must not be set for variant "{variant}".'
+                _(f'"original" must not be set for variant "{variant}".')
```
Comment (anehx) [inline alexandria/core/serializers.py:218]: Same here

File: alexandria/core/serializers.py [hunk 6]
```diff
@@ -206,26 +208,34 @@ def validate(self, *args, **kwargs):
...
+        category = models.Document.objects.get(
+            pk=self.initial_data.get("document")["id"]
+        ).category
+        if (
+            category.allowed_mime_types is not None
+            and len(category.allowed_mime_types)
+            and mime_type not in category.allowed_mime_types
+        ):
+            raise ValidationError(
+                _(f"File type {mime_type} is not allowed in category {category.pk}.")
```
Comment (anehx) [inline alexandria/core/serializers.py:236]: Same here

File: alexandria/core/serializers.py (hunk 4 above)
Comment (Yelinz) [reply alexandria/core/serializers.py:215]: didnt know, makes sense. Fixes

File: alexandria/core/serializers.py (hunk 5 above)
Comment (Yelinz) [reply alexandria/core/serializers.py:218]: Fixed

File: alexandria/core/serializers.py (hunk 6 above)
Comment (Yelinz) [reply alexandria/core/serializers.py:236]: Fixed

**Review by czosel — APPROVED** [review]
LGTM :+1:

**Pull request merged**